src/sql-parser/tests/testdata/scalar

```diff
@@ -83,11 +83,7 @@ Op { op: "!~*", expr1: Identifier([Ident("a")]), expr2: Some(Value(String("foo")
 parse-scalar
 a !x 'foo'
 ----
-error:
-Parse error:
-a !x 'foo'
-     ^^^^^
-extra token after expression
+Op { op: "!", expr1: Identifier([Ident("a")]), expr2: Some(Cast { expr: Value(String("foo")), data_type: Custom("x") }) }
```

Comment:
this seems worse, still wip?

Reply:
Nope! This is the correct way to parse this statement per Postgres. The old code was just getting it wrong.